core/src/main/java/org/elasticsearch/index/query/MatchQueryParser.java

```diff
@@ -69,8 +64,18 @@ public Query parse(QueryShardContext context) throws IOException, QueryParsingEx
 
         Object value = null;
         float boost = AbstractQueryBuilder.DEFAULT_BOOST;
-        MatchQuery matchQuery = new MatchQuery(context);
         String minimumShouldMatch = null;
+        String analyzer = null;
```

Comment:
I'm not sure if there's a reasonable default for the analyzer setting. If this option is not set (null) then later we try to resolve it by using the field type of fall back to the search analyzer in the MapperService. Will leave this as null for now since I don't know how we otherwise indicate that the user hasn't specified anything explicitely